lib/timers.js

```diff
@@ -317,51 +322,76 @@ exports.enroll = function(item, msecs) {
  */
 
 
-exports.setTimeout = function(callback, after) {
+exports.setTimeout = function(callback, after, arg1, arg2, arg3) {
   if (typeof callback !== 'function') {
     throw new TypeError('"callback" argument must be a function');
   }
 
-  after *= 1; // coalesce to number or NaN
-
-  if (!(after >= 1 && after <= TIMEOUT_MAX)) {
-    after = 1; // schedule on next tick, follows browser behaviour
+  var len = arguments.length;
+  var args;
+  if (len === 3) {
+    args = [arg1];
+  } else if (len === 4) {
+    args = [arg1, arg2];
+  } else if (len > 4) {
+    args = [arg1, arg2, arg3];
+    for (var i = 5; i < arguments.length; i++)
+      // extend array dynamically, makes .apply run much faster in v6.0.0
+      args[i - 2] = arguments[i];
   }
 
-  var timer = new Timeout(after);
-  var length = arguments.length;
-  var ontimeout = callback;
-  switch (length) {
-    // fast cases
-    case 1:
-    case 2:
-      break;
-    case 3:
-      ontimeout = () => callback.call(timer, arguments[2]);
-      break;
-    case 4:
-      ontimeout = () => callback.call(timer, arguments[2], arguments[3]);
-      break;
-    case 5:
-      ontimeout =
-        () => callback.call(timer, arguments[2], arguments[3], arguments[4]);
-      break;
-    // slow case
-    default:
-      var args = new Array(length - 2);
-      for (var i = 2; i < length; i++)
-        args[i - 2] = arguments[i];
-      ontimeout = () => callback.apply(timer, args);
-      break;
-  }
-  timer._onTimeout = ontimeout;
+  return createSingleTimeout(callback, after, args);
+};
 
-  if (process.domain) timer.domain = process.domain;
+function createSingleTimeout(callback, after, args) {
+  after *= 1; // coalesce to number or NaN
+  if (!(after >= 1 && after <= TIMEOUT_MAX))
+    after = 1; // schedule on next tick, follows browser behaviour
+
+  var timer = new Timeout(after, callback, args);
+  if (process.domain)
+    timer.domain = process.domain;
 
   active(timer);
 
   return timer;
-};
+}
+
+
+function ontimeout(timer) {
+  var args = timer._timerArgs;
+  var callback = timer._onTimeout;
+  if (!args)
+    callback.call(timer);
+  else {
+    switch (args.length) {
+      case 1:
+        callback.call(timer, args[0]);
+        break;
+      case 2:
+        callback.call(timer, args[0], args[1]);
+        break;
+      case 3:
+        callback.call(timer, args[0], args[1], args[2]);
+        break;
+      default:
+        callback.apply(timer, args);
+    }
+  }
+  if (timer._repeat != null)
+    rearm(timer);
+}
+
+
+function rearm(timer) {
+  // If timer is unref'd (or was - it's permanently removed from the list.)
+  if (timer._handle && timer instanceof Timeout) {
```

Comment:
Unfortunately not, as `_http_outgoing` and `net` have a `._handle` and manually `enroll()` pre-made objects. Node will crash since those `._handle` instances are not only not `Timeout` instances, but they also do not have a `.start()` method.